_opensearch/rest-api/snapshots/create-repository.md

```diff
@@ -0,0 +1,101 @@
+---
+layout: default
+title: Create Snapshot Repository
+parent: Snapshot APIs
+grand_parent: REST API reference
+nav_order: 1
+---
+
+## Create snapshot repository
+
+Registers a new repository to store snapshots. Repositories comprise two types:
+
+* File system (`fs`)
+
+* Amazon S3 bucket (`s3`)
+
+To learn more about repositories, see [Register repository]({{site.url}}{{site.baseurl}}/opensearch/snapshots/snapshot-restore#register-repository).
+
+
+### Path parameters
+
+Parameter | Data Type | Description
+:--- | :--- | :---
+repository | String | Repostory name. |
+
+
+### Query parameters
+
+None
+
+#### Sample request
+
+You can create repositories with `POST` or `PUT`. 
+{: .note}
+
+The following requests create a repository called `my-first-repo`:
+
+`POST _snapshot/my-first-repo/`
```

Comment:
This request is not going to work as-is because location is a required parameter. If a user tries it, he/she will get an error. Please update this request after testing it.

Reply:
Thank you, kolchafa. (BTW, it would be nice to address you by your name if you care to share it. :) )

Hi @Naarcha-AWS! Need to discuss this. The [STYLE_API_TEMPLATE.md](https://github.com/opensearch-project/documentation-website/blob/main/STYLE_API_TEMPLATE.md) file indicates to show just the HTTP method and the endpoint/path. For a request like this one, a request body is needed, the template indicates to discuss the request body later in the `Request body` section. I'm not sure how to bridge that gap. Thanks.

Aside: I tested the API using a request body.

Reply:
The basic issue has been resolved with the new Style API Template. Nate told me not to retrofit this file. In the meantime, I've added extra text to explain the requirement for a request body.